dashboard/lib/build_dashboard_page.dart

```diff
@@ -79,6 +85,32 @@ class BuildDashboardPageState extends State<BuildDashboardPage> {
             child: Column(
               mainAxisSize: MainAxisSize.min,
               children: <Widget>[
+                DropdownButton<String>(
+                  value: _buildState.currentRepo,
+                  icon: const Icon(
+                    Icons.arrow_downward,
+                  ),
+                  iconSize: 24,
+                  elevation: 16,
+                  underline: Container(
+                    height: 2,
+                  ),
+                  onChanged: (String repo) {
+                    final BuildDashboardArguments args =
+                        BuildDashboardArguments(repo: repo, branch: _buildState.currentBranch);
+                    Navigator.pushNamed(
+                      context,
+                      args.toRoute(),
+                      arguments: args,
```

Comment:
So, the branch and repository information get passed twice? Once in the name of the route and once as arguments object?

Reply:
It looks like "onGenerateRoute" just ignores the arguments, though?

Reply:
Is there a better way of passing the args from onGenerateRoute? I'm doing

```
                  repo: uriData.queryParameters['repo'],
                  branch: uriData.queryParameters['branch'],
```

to initialize the fields based on what was passed from `RouteSettings`

Reply:
I removed the settings, but I still needed to pass it in `onGenerateRoute` so the URL would be updated